### ISSUE TITLE
check for source control

### ISSUE THREAD

**Issue by benmccann** [svelte-add/svelte-add#456]
### Topic

cli

### Description

I've run it on projects several times without creating a git project and have regretted doing so. It's really nice to have the before after

### Describe the solution you'd like

bring back the git precondition instead of running it only when there's a git repo. allow the user to skip it if they'd like. we should probably also check for other known source control types like subversion, etc.

**Comment by manuel3108** [comment]
> When it's a dirty git repo we should warn them of that immediately rather than having them choose the options first and then telling them

What do you mean by choosing the options first and then telling them? iirc, pre-conditions are executed after the adder selection. Do you mean they should be executed as first step, before selecting the adders?

What would we do for adder specific preconditions? We cannot run them, before the user has decided which adders he want's to add.

**Comment by benmccann** [comment]
Yes, I meant that since we already know it's an issue we shouldn't have to wait for the user to make selections. We could do this as a special check outside preconditions or split preconditions into two different hooks like `onStart` and `onAdderSelection` or something
